RecoTracker/LST/plugins/LSTOutputConverter.cc

```diff
@@ -177,8 +177,8 @@ void LSTOutputConverter::produce(edm::Event& iEvent, const edm::EventSetup& iSet
         return true;
       } else if (GeomDetEnumerators::isOuterTracker(asub) && GeomDetEnumerators::isInnerTracker(bsub)) {
         return false;
-      } else if (asub != bsub) {
-        return asub < bsub;
```

Comment:
The comparisons returned no changes.
I guess I mis-read `GeomDetEnumerators GeomDet::subDetector()` order implied here `TIB = 2,    TOB = 3,    TID = 4,    TEC = 5`
vs the Phase2 hit detId, which comes from `StripSubdetector` and has `TID = 4, TOB = 5`.

This, however does not explain why in the mkFit ntuple test the order of the hits is apparently incorrect